app/src/main/java/com/capstone/weatherapp/CityListViewModel.kt

```diff
@@ -1,26 +1,45 @@
 package com.capstone.weatherapp
 
+import androidx.lifecycle.LiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModelProvider
 import androidx.lifecycle.viewModelScope
 import kotlinx.coroutines.launch
 
 private const val CITIES = "4772354,5368361,5128581,4887398,4699066,5809844,5856195,5419384,5391959,4684888,5746545,4990729,4335045"
 private const val APIKEY = "6849ae760f417fb8188f4bb7fd0d92fc"
 private const val UNITS = "imperial"
 
-class CityListViewModel : ViewModel() {
-    private val _cityList = MutableLiveData<List<City>>()
+class CityListViewModel(private val weatherRepo: WeatherRepository) : ViewModel() {
+    private val _cityList = weatherRepo.cityListData
     val cityList: MutableLiveData<List<City>>
```

Comment:
My b for not catching this earlier but this should not be mutable, should just be LiveData since we don't want the fragment or any UI to be able to update this value

Reply:
Actually, you don't even need the private _cityList anymore, you should just need a public cityList that's set to weatherRepo.cityListData. Since you're just setting it to the value of another variable from the repository, don't need the private backing variable anymore